simulation/traffic_simulator/src/api/api.cpp

```diff
@@ -397,4 +423,40 @@ void API::requestLaneChange(
   entity_manager_ptr_->requestLaneChange(name, target, trajectory_shape, constraint);
 }
 
+void API::requestSpeedChange(const std::string & name, double target_speed, bool continuous)
+{
+  assert(entity_manager_ptr_);
+  entity_manager_ptr_->requestSpeedChange(name, target_speed, continuous);
+  if (entity_manager_ptr_->isEgo(name) and not isNpcLogicStarted()) {
+    ego_entity_simulation_->requestSpeedChange(target_speed);
+  }
+}
+
+void API::requestSpeedChange(
+  const std::string & name, const double target_speed, const speed_change::Transition transition,
+  const speed_change::Constraint constraint, const bool continuous)
+{
+  assert(entity_manager_ptr_);
```

Comment:
It is recommended to throw an exception, instead of using `assert`.

Reply:
I've changed it according to your feedback -> a0e78a1744acd16f664c71bfc9006bdadd22baee